### PR TITLE
Add Vault CI test with deployed Vault with Transit

### PR DESCRIPTION
**Pull request by shtripat**
This PR introduces a github workflow to deploy HashiCorp Vault with Transit enabled and execute tests.

### CODE REVIEW

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+jobs:
+  valut-ci-test:
+    name: Test Vault
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.21.4
```
Comment (harshavardhana) [inline .github/workflows/ci-test.yml:16]:
```suggestion
        go-version: 1.21.5
```

File: .github/workflows/ci-test.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: CI tests
+
+on:
+  push:
```
Comment (harshavardhana) [inline .github/workflows/ci-test.yml:4]:
```suggestion
  pull_request:
```

File: .github/workflows/ci-test.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: CI tests
```
Comment (harshavardhana) [inline .github/workflows/ci-test.yml:1]:
```suggestion
name: Vault Integration CI tests
```

**Comment by harshavardhana** [comment]
PTAL at the CI failure

```
go: downloading github.com/go-logr/stdr v1.2.2
=== RUN   TestVaultCI
    vault_ci_test.go:21: kesconf: invalid vault keystore: invalid approle config: no approle ID specified
--- FAIL: TestVaultCI (0.00s)
FAIL
FAIL	github.com/minio/kes/kesconf	0.014s
FAIL
Error: Process completed with exit code 1.
```

**Comment by harshavardhana** [comment]
Looks like KV List permissions are missing @shtripat

**Comment by shtripat** [comment]
> Looks like KV List permissions are missing @shtripat

Somehow locally inside a ubuntu VM with below kes-policy.hcl the tests run fine
```
path "transit/encrypt/my-key" {
   capabilities = [ "update" ]
}
path "transit/decrypt/my-key" {
   capabilities = [ "update" ]
}
path "kv/*" {
   capabilities = [ "create", "read", "delete", "list" ]
}
```

```
root@vm-01:~/kes# go test ./kesconf -v -vault-ci.config=/root/kes/kesconf/testdata/vault/kes-config-vault.yml -run="TestVaultCI"
=== RUN   TestVaultCI
=== RUN   TestVaultCI/Create
=== RUN   TestVaultCI/Get
=== RUN   TestVaultCI/Status
--- PASS: TestVaultCI (0.05s)
    --- PASS: TestVaultCI/Create (0.01s)
    --- PASS: TestVaultCI/Get (0.03s)
    --- PASS: TestVaultCI/Status (0.00s)
PASS
ok  	github.com/minio/kes/kesconf	0.064s
```

do you mean to change policy as below?

```
path "transit/encrypt/my-key" {
   capabilities = [ "update", "list" ]
}
path "transit/decrypt/my-key" {
   capabilities = [ "update", "list" ]
}
path "kv/*" {
   capabilities = [ "create", "read", "delete", "list" ]
}
```

**Review by aead — DISMISSED** [review]

**Comment by bh4t** [comment]
@harshavardhana @aead can one of you PTAL?

**Comment by harshavardhana** [comment]
This is close to a year old, looks like this may never get merged. Closing.

**Comment by harshavardhana** [comment]
Feel free to send when its ready.

**Pull request closed without merging**